libs/langchain/tests/integration_tests/vectorstores/test_pinecone.py

```diff
@@ -231,3 +231,31 @@ def test_relevance_score_bound(self, embedding_openai: OpenAIEmbeddings) -> None
         assert all(
             (1 >= score or np.isclose(score, 1)) and score >= 0 for _, score in output
         )
+
+    @pytest.mark.skipif(reason="slow to run for benchmark")
+    def test_from_texts_with_metadatas_benchmark(
+        self, documents: List[Document], embedding_openai: OpenAIEmbeddings
+    ) -> None:
+        """Test end to end construction and search."""
+        POOL_THREADS = 4
```

Comment:
Does it make sense to set up a dict or similar containing several values for each of these params and then running through each in a test? To essentially flex more potential combinations to flush out any potential issues. LMK if this doesn't make sense.

Reply:
You mean like to use parametrized tests ie something like this:
```
    @pytest.mark.parametrize(
        "pool_threads,batch_size,embeddings_chunk_size,multipier",
        [(1, 32, 32, 1000),  # simulate single threaded with embeddings_chunk_size = batch_size = 32
         (1, 32, 1000, 1000),  # simulate single threaded with embeddings_chunk_size = 1000
         (4, 32, 1000, 1000),  # simulate 4 threaded with embeddings_chunk_size = 1000
         (20, 64, 5000, 1000)]  # simulate 20 threaded with embeddings_chunk_size = 5000
    )
```

Reply:
Yup! Exactly what I was thinking - coming from Golang I'd call them table tests, not sure what / if there's a common equivalent in the Python world.